dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/report/TrackerTypeReport.java

```diff
@@ -92,20 +86,16 @@ public boolean isEmpty() {
   }
 
   public void addEntity(Entity entity) {
-    this.entityReport.put(entity.getIndex(), entity);
+    this.entityReport.add(entity);
   }
 
   private List<Error> getErrorReports() {
     List<Error> errorReports = new ArrayList<>();
-    entityReport.values().forEach(entity -> errorReports.addAll(entity.getErrorReports()));
+    entityReport.forEach(entity -> errorReports.addAll(entity.getErrorReports()));
```

Comment:
I think we could also do

```suggestion
  return entityReport.values().stream().flatMap(e -> e.getErrorReports().stream()).toList();
```